lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -227,6 +227,33 @@ static LogicalResult verifyMemoryAccessAttribute(LoadStoreOpTy loadStoreOp) {
   return success();
 }
 
+template <typename BarrierOp>
+static LogicalResult verifyMemorySemantics(BarrierOp op) {
+  auto memorySemantics = op.memory_semantics();
+  SmallVector<spirv::MemorySemantics, 4> memorySemanticsBits{
+      spirv::MemorySemantics::Acquire, spirv::MemorySemantics::Release,
+      spirv::MemorySemantics::AcquireRelease,
+      spirv::MemorySemantics::SequentiallyConsistent};
+
+  uint32_t bitCount = 0;
```

Comment:
I think we can do ```
auto atMostOneInSet = spirv::MemorySemantics::... | ...;

auto bitCount = llvm::countPopulation(memorySemantics & atMostOneInSet);
```

We need to generate `operator&` for bit enums to do the above; but it should be fairly straightforward: https://github.com/tensorflow/mlir/blob/master/tools/mlir-tblgen/SPIRVUtilsGen.cpp#L707

Reply:
@antiagainst thanks, I'll update it!